### ISSUE TITLE
InCargoBay is always set to true by SetCargoBays, meaning custom bay toggles never fire.

### ISSUE THREAD

**Issue by nisamson** [BrettRyland/BDArmory#329]
The "In Cargo Bay" option overrides the "Custom Bay Toggle" setting, but this is not documented in the Wiki or in the GUI (probably the best place for it).

https://github.com/BrettRyland/BDArmory/blob/e29bed9c431420028f96ffc7c5706525edfa9d1b/BDArmory/Modules/MissileFire.cs#L2779-L2838

While it may not be *necessary* to ever set both, it's also not obvious that setting both won't do what the user probably thinks it will. I was only able to figure out why my custom toggle wasn't activating by reading the source code.

**Comment by nisamson** [comment]
I'm investigating to see if it might make more sense to have "Custom Bay Toggle" override "In Cargo Bay" instead.

**Comment by nisamson** [comment]
It looks like there's a deeper bug here.

https://github.com/BrettRyland/BDArmory/blob/e29bed9c431420028f96ffc7c5706525edfa9d1b/BDArmory/Modules/MissileFire.cs#L2769-L2777

This section of the code sets "InCargoBay" to true for all shielded missiles.

That makes this if statement *always* be entered, even if the custom cargo bay toggle is set.
https://github.com/BrettRyland/BDArmory/blob/e29bed9c431420028f96ffc7c5706525edfa9d1b/BDArmory/Modules/MissileFire.cs#L2779

I recommend editing line 2779 to read instead:

```cs
if (CurrentMissile.inCargoBay && CurrentMissile.customBayGroup == "0") {
```

***OR***, and I think this is the better option, switch that branch with the next one, reading instead:

```cs
                if (uint.Parse(CurrentMissile.customBayGroup) > 0) // Missile uses a custom bay, open it to fire 
                {
                    ...
                }
                else if (CurrentMissile.inCargoBay)
                {
                    ...
                }
                else
```

This second option has the advantage of removing a redundant check.

**Comment by BrettRyland** [comment]
Yep, fixed by merging #330
